raiden-ts/src/transport/epics/presence.ts

```diff
@@ -262,3 +250,28 @@ export const matrixMonitorChannelPresenceEpic = (
     filter(channelMonitor.is),
     map((action) => matrixPresence.request(undefined, { address: action.meta.partner })),
   );
+
+/**
+ * Update our matrix's avatarUrl on config.caps changes
+ *
+ * @param action$ - Observable of RaidenActions
+ * @param state$ - Observable of RaidenStates
+ * @param deps - Epics dependencies
+ * @returns Observable which never emits
+ */
+export const matrixUpdateCapsEpic = (
+  {}: Observable<RaidenAction>,
+  {}: Observable<RaidenState>,
+  { matrix$, config$ }: RaidenEpicDeps,
+): Observable<never> =>
+  config$.pipe(
+    pluckDistinct('caps'),
+    skip(1), // skip replay(1) and act only on changes
+    mergeMap((caps) => matrix$.pipe(map((matrix) => [caps, matrix] as const))),
+    mergeMap(async ([caps, matrix]) =>
+      matrix.setAvatarUrl(caps && !isEmpty(caps) ? stringifyCaps(caps) : '').catch(() => {
+        /* ignore http errors */
+      }),
+    ),
```

Comment:
Possible unnecessary `async` here if we don't await anything

Reply:
Some plugin on my editor suggested it when I had a function with `return`, I think even without `await`, it prefers functions returning promises to be marked as `async`